pkg/providers/instance/types.go

```diff
@@ -27,28 +27,39 @@ import (
 // Instance is an internal data representation of either an ec2.Instance or an ec2.FleetInstance
 // It contains all the common data that is needed to inject into the Machine from either of these responses
 type Instance struct {
-	LaunchTime       time.Time
-	State            ec2types.InstanceStateName
-	ID               string
-	ImageID          string
-	Type             ec2types.InstanceType
-	Zone             string
-	CapacityType     string
-	SecurityGroupIDs []string
-	SubnetID         string
-	Tags             map[string]string
-	EFAEnabled       bool
+	LaunchTime            time.Time
+	State                 ec2types.InstanceStateName
+	ID                    string
+	ImageID               string
+	Type                  ec2types.InstanceType
+	Zone                  string
+	CapacityType          string
+	CapacityReservationID string
+	SecurityGroupIDs      []string
+	SubnetID              string
+	Tags                  map[string]string
+	EFAEnabled            bool
 }
 
 func NewInstance(out ec2types.Instance) *Instance {
 	return &Instance{
-		LaunchTime:   aws.ToTime(out.LaunchTime),
-		State:        out.State.Name,
-		ID:           aws.ToString(out.InstanceId),
-		ImageID:      aws.ToString(out.ImageId),
-		Type:         out.InstanceType,
-		Zone:         aws.ToString(out.Placement.AvailabilityZone),
-		CapacityType: lo.Ternary(out.SpotInstanceRequestId != nil, karpv1.CapacityTypeSpot, karpv1.CapacityTypeOnDemand),
+		LaunchTime: aws.ToTime(out.LaunchTime),
+		State:      out.State.Name,
+		ID:         aws.ToString(out.InstanceId),
+		ImageID:    aws.ToString(out.ImageId),
+		Type:       out.InstanceType,
+		Zone:       aws.ToString(out.Placement.AvailabilityZone),
+		CapacityType: func() string {
```

Comment:
`lo.If` I believe is another way to do this because there's also `lo.ElseIf` off of that -- that change is really stylistic so it's really up to you if you want to use this or keep the current function syntax